src/Sarif/Baseline/V2/WhatComparer.cs

```diff
@@ -0,0 +1,192 @@
+        {
+            if (result?.Result == null) { yield break; }
+
+            // Add Guid
+            if (result.Result.Guid != null)
+            {
+                yield return new WhatComponent(result.RuleId, PropertySetBase, "Guid", result.Result.Guid);
+            }
+
+            // Add Message Text
+            string messageText = result.Result.GetMessageText(result.Result.GetRule(result.OriginalRun));
+            if (!string.IsNullOrEmpty(messageText))
+            {
+                yield return new WhatComponent(result.RuleId, PropertySetBase, "Message", messageText);
+            }
+
+            // Add each Fingerprint
+            if (result.Result.Fingerprints != null)
+            {
+                foreach (var fingerprint in result.Result.Fingerprints)
+                {
+                    yield return new WhatComponent(result.RuleId, "Fingerprint", fingerprint.Key, fingerprint.Value);
+                }
+            }
+
+            // Add each PartialFingerprint
+            if (result.Result.PartialFingerprints != null)
+            {
+                foreach (var fingerprint in result.Result.PartialFingerprints)
+                {
+                    yield return new WhatComponent(result.RuleId, "PartialFingerprint", fingerprint.Key, fingerprint.Value);
+                }
+            }
+
+            if (result.Result.Locations != null)
+            {
+                foreach (Location location in result.Result.Locations)
+                {
+                    string snippet = location?.PhysicalLocation?.Region?.Snippet?.Text;
+                    if (!string.IsNullOrEmpty(snippet))
+                    {
+                        yield return new WhatComponent(result.RuleId, PropertySetBase, "Location.Snippet", snippet);
+                    }
+                }
+            }
+
+            // Add each Property
+            if (result.Result.Properties != null)
+            {
+                foreach (var property in result.Result.Properties)
+                {
+                    yield return new WhatComponent(result.RuleId, "Property", property.Key, property.Value.SerializedValue);
+                }
+            }
+        }
+
+        /// <summary>
+        ///  Match the 'What' properties of two ExtractedResults.
+        /// </summary>
+        /// <param name="right">ExtractedResult to match</param>
+        /// <returns>True if *any* 'What' property matches, False otherwise</returns>
+        public static bool MatchesWhat(ExtractedResult left, ExtractedResult right)
+        {
+            if (left?.Result == null || right?.Result == null) { return false; }
+
+            // Match Guid
+            if (left.Result.Guid != null)
+            {
+                if (string.Equals(left.Result.Guid, right.Result.Guid))
+                {
+                    return true;
+                }
+
+                // Non-Match doesn't force false, as GUIDs are often re-generated by tools.
```

Comment:
>GUIDs are often re-generated by tools [](start = 53, length = 37)

I suppose we can't stop tools from doing that, but they're not supposed to. A GUID is supposed to identify an "instance" of a result for all time.